src/main/java/net/mcreator/workspace/references/ReferencesFinder.java

```diff
@@ -0,0 +1,286 @@
+	 * @return List of all generatable elements contained in the provided workspace.
+	 */
+	private static List<GeneratableElement> getGeneratableElements(Workspace workspace) {
+		return workspace.getModElements().stream().map(ModElement::getGeneratableElement).collect(Collectors.toList());
+	}
+
+	public static Set<ModElement> searchModElementUsages(Workspace workspace, ModElement element) {
+		//@formatter:off
+		return getGeneratableElements(workspace).parallelStream()
+			.filter(ge ->
+				anyValueMatches(ge, String.class, e -> e.isAnnotationPresent(ModElementReference.class), (a, t) -> {
+					ModElementReference ref = a.getAnnotation(ModElementReference.class);
+					return !Set.of(ref.defaultValues()).contains(t) && element.getName().equals(
+							GeneratorWrapper.getElementPlainName(t));
+				}) ||
+				anyValueMatches(ge, MappableElement.class, e -> e.isAnnotationPresent(ModElementReference.class), (a, t) ->
+					element.getName().equals(GeneratorWrapper.getElementPlainName(t.getUnmappedValue()))
+				) ||
+				anyValueMatches(ge, Procedure.class, e -> e.isAnnotationPresent(ModElementReference.class), (a, t) ->
+					element.getName().equals(t.getName())
+				) ||
+				anyValueMatches(ge, String.class, e -> e.isAnnotationPresent(BlocklyXML.class), (a, t) ->
+					t.contains(">CUSTOM:" + element.getName() + "</field>") || t.contains(">" + element.getName() + "</field")
+				)
+			)
+			.map(GeneratableElement::getModElement).filter(me -> !me.equals(element)).collect(Collectors.toSet());
+		//@formatter:on
+	}
+
+	public static Set<ModElement> searchTextureUsages(Workspace workspace, File texture, TextureType type) {
+		return getGeneratableElements(workspace).parallelStream().filter(ge -> anyValueMatches(ge, String.class, e -> {
+			TextureReference ref = e.getAnnotation(TextureReference.class);
+			return ref != null && ref.value() == type;
+		}, (a, t) -> {
+			TextureReference ref = a.getAnnotation(TextureReference.class);
+			if (!Set.of(ref.defaultValues()).contains(t)) {
+				for (String template : ref.files()) {
+					String file = template.isEmpty() ? t : template.formatted(t);
+					if (workspace.getFolderManager().getTextureFile(FilenameUtilsPatched.removeExtension(file), type)
+							.equals(texture))
+						return true;
+				}
+			}
+			return false;
+		})).map(GeneratableElement::getModElement).collect(Collectors.toSet());
+	}
+
+	public static Set<ModElement> searchModelUsages(Workspace workspace, Model model) {
+		return getGeneratableElements(workspace).parallelStream().filter(ge -> anyValueMatches(ge, Model.class, e -> {
+					ResourceReference ref = e.getAnnotation(ResourceReference.class);
+					return ref != null && ref.value().equals("model");
+				}, (a, t) -> model.equals(t) || TexturedModel.getModelTextureMapVariations(model).contains(t)))
+				.map(GeneratableElement::getModElement).collect(Collectors.toSet());
+	}
+
+	public static Set<ModElement> searchSoundUsages(Workspace workspace, SoundElement sound) {

```

Comment:
Sounds can also be used in procedure blocks so procedures should be checked too here